src/main/java/harmonised/pmmo/mixin/EnchantApplyMixin.java

```diff
@@ -2,32 +2,32 @@
 
 import java.util.List;
 
-import org.spongepowered.asm.mixin.Debug;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.Level;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
 
 import harmonised.pmmo.api.events.EnchantEvent;
-import net.minecraft.core.BlockPos;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.inventory.EnchantmentMenu;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.enchantment.EnchantmentInstance;
-import net.minecraft.world.level.Level;
 import net.minecraftforge.common.MinecraftForge;
 
-@Debug
 @Mixin(EnchantmentMenu.class)
-public class EnchantApplyMixin {	
-
+public class EnchantApplyMixin {
 	@Inject(method = "lambda$clickMenuButton$1(Lnet/minecraft/world/item/ItemStack;ILnet/minecraft/world/entity/player/Player;ILnet/minecraft/world/item/ItemStack;Lnet/minecraft/world/level/Level;Lnet/minecraft/core/BlockPos;)V",
 	        at = @At(
-	        	value = "INVOKE", 
-	        	target = "Lnet/minecraft/world/item/ItemStack;enchant(Lnet/minecraft/world/item/enchantment/Enchantment;I)V"), 
-	        locals = LocalCapture.CAPTURE_FAILSOFT)
-	private void projectmmo$$enchantHandle(ItemStack $$stack, int $$1x, Player player, int $$3x, ItemStack $$4x, Level $$5, BlockPos $$6, CallbackInfo ci, ItemStack stack, List<?> $$8, boolean $$9, int $$11, EnchantmentInstance instance) {
-		MinecraftForge.EVENT_BUS.post(new EnchantEvent(player, stack, instance));
- 	}
+	        	value = "INVOKE",
+	        	target = "Lnet/minecraft/advancements/critereon/EnchantedItemTrigger;trigger(Lnet/minecraft/server/level/ServerPlayer;Lnet/minecraft/world/item/ItemStack;I)V"),
```

Comment:
This entry point would only execute on the server.  addons or users who create perks based on the enchant event being on the client would no longer be able to have their client-side perks activate on this event.

If you're going to capture locals anyways and risk the local_capture_fail at runtime, there's no reason to change the target for this mixin.

Reply:
Damn, you are right. I've got tricked by the event handler, thinking it would be a welcome side-effect since it's logic is behind server-side checks anyway... again, forgetting this event is a part of pmmo's API. It's an easy fix though, I'll just move the target two lines higher to awardStat() call.

As for crash on capture fail - it's very much done on purpose. This is a part of core functionality - if it's broken then the mod is broken. And I would like to be sure it will get noticed when that happens.

Reply:
sorry, i didn't mean to imply that the failhard on locals was undesired.  you are correct, that should remain for the reason you outline.  

That said, we could probably move the mixin even further up and reduce our locals even further.  Since there is no exit condition inside of the lambda for `this.access.execute` and the instatiating logic for `list` uses class level variables, we could invoke inside the `else` before the `execute` call.  If we use an access widener on `this.enchantSlots` we could also probably drop locals entirely and just recapture the `itemstack` local inside of our mixin implementation.

I am actually going to test that now.  that would unfortunately make this PR obsolete, but you've piqued my interest